### DISCUSSION TITLE
Immersion/flavour-themed additions for the Outposts (Campaign)

### DISCUSSION THREAD

**Discussion by IgRAzm** [FakeFishGames/Barotrauma#8297] [Ideas and suggestions]
There are many different settlement types in the game already, and it's very cool that the layouts are easily moddable, but there are still a few of unexplored possibilities in my opinion.

   Not yet docked outposts:
Currently in game if you move to an outpost without docking, it doesn't really assist you in any way, in fact you can't enter it as the hatch won't open until the submarine is docked (obviously the settlement area isn't even loaded yet). But this outer part of an outpost doesn't need to know all about the structure to just correspond to the basic properties of this location. What I mean is the outer area would be determined together with the level generation and get different functions and layouts based on the randomized layout it chose. As such, outposts with tiers above a habitation receive sonars, as well as periscopes with projectors, scaning the surrounding area and focusing on the creatures. The military and colony outposts instead get actual defensive turrets that fend off enemies following our submarine.

   World details:
The current date (either randomized year or probably better to have it as 21XX year) is displayed in the loading screens and tab menus, so that players can keep track of the diegetic Earth's Solar cycle; each character has a marked date for when they first joined the crew. The outpost/interest point login screen would be the only thing actually progressing that time rather than the real time; depending on the amount of things you bought, upgraded and switched, it can take from a day to a couple, while for empty locations and such it's less than a day. The time of day wouldn't be shown, which seems to be most fitting for a place without any significant daylight.

A majority of the outposts receive unaccessible areas to the edges, that have signs for player to not be confused, such as striped lines and security posts; sometimes npcs spawn in there and get into/out of the playable area (sometimes even ai-crews on standby between travels just like you), creating  an illusion of a larger world in the settlement (also the navigation panel visual of the outpost appears differently, maybe showing a layout of outpost instead of an outline of where it ends). The AIs of the colonies will occasionally turn on an alarm when they encounter packs of crawlers and such, shooting them down with turrets in semi-scripted events. Most importantly I think, having people with different physique and age would add big deal to the liveliness of the outposts: older, somewhat wisened people with hunched backs, fat people, younger teenagers and maybe even children (game isn't too expressive on gory realism when it comes to human vs human, so honestly I don't see why not; can always just make them invincible and simply alerting the security though). After all, even a harsh world of Europa got inhabitants not fit for crewing a submarine; hardly all scientists and miners (they mine uranium and such in likely much less stable and comfortable work environment than today) are expected to be strong of body, while every single character we control has been too young for sailing once - they didn't land on Europa yesterday after all.

   Dockyard Outposts:
Usually very big and with rarity depending on your chosen vessel size (with the larger vessels they are formed only out of colonies), Dockyards are outposts with a procedurally generated outside area each; unlike when docking to the other outposts, here your submarine has a wide encompassing space for parking; when you approach it it will make docking possible if the sub is simply inside in its entirety. After loading it is drained of water, has accessible convenience terminal near the bottom for you to access all of Dockyard's features (even including submarine switch and upgrade features, that however only apply after undocking), populated with workers and security, and filled with scaffoldings, stairs and ladders (with more rough and cheap appearance than the normal indoors ones). The docks have space for both your sub and all its supporting vessels, affixed to the scaffoldings. For a cost you can get temporary access to hands-only, hosepiped and station-powered tools like heavy welding tool, heavy plasma cutter and heavy pulverizer. The former two give players option to repair and rewire the entire submarine without any trouble, while the third comes in two variants: to either set specific targeted objects of the submarine to any color you choose, or to wash the paint away and return them to the editor-set submarine coloring (as well as removing regular spray paint and liquids on top). Like normal outposts, it comes with an upper area, but you can go there only by accessing one of terminals and calling an elevator (it happens through another loading screen, making the Dockyards basically two-layered outposts).

**Comment by DKAMX** [comment]
Your idea is cool, I also came up with an idea that they should add a new type of outpost similar to the beacon.
They were built on the sea bed and have an upper docking port so submarines like Orca or other player designs can be widely used.
Hope the devs will adopt this and update it in the future.
